File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
import heapq
def solution(participant, completion):
    heapq.heapify(completion)
    heapq.heapify(participant)
    ans = ""
    while participant:
        cur_part = heapq.heappop(participant)
        if completion:
            cur_com = heapq.heappop(completion)
            if cur_part != cur_com:
                ans = cur_part
                break
            else:
                continue
        else:
            ans = cur_part
            break
        
        
    return ans

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.io.*;
class Solution {
    static Map<String,Integer> partMap;
    static Map<String,Integer> compMap;
    static StringBuilder sb = new StringBuilder();
    
    public String solution(String[] participant, String[] completion) {
        partMap = new HashMap<>();
        compMap = new HashMap<>();
        for(String done:completion){
            compMap.put(done,compMap.getOrDefault(done,0)+1);
        }
        for(String sportMan:participant){
            partMap.put(sportMan,partMap.getOrDefault(sportMan,0)+1);
        }
        for(String man:partMap.keySet()){
            if(compMap.getOrDefault(man,0)<partMap.get(man)){
                sb.append(man);
            }
        }
        return sb.toString();
    }
    public static void main(String[] args){
        Solution s = new Solution();
        System.out.println(s);
    }
}